cmd/clusterctl/client/alpha/cluster.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package alpha
+
+import (
+	"context"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster"
+)
+
+// getCluster retrieves the Cluster object corresponding to the name and namespace specified.
+func getCluster(ctx context.Context, proxy cluster.Proxy, name, namespace string) (*clusterv1.Cluster, error) {
```

Comment:
Is it worth moving this to a util file or the util package? I'd imagine there are lots of instances where we would want to get a cluster outside of clusterctl.

Reply:
I removed `alpha/cluster.go` and changed to use `util.GetClusterByName` instead. I forgot why I re-implement `getCluster` at that moment I created this PR, but it still passed e2e-test after replacing it with `GetClusterByName.